openeo/rest/auth/config.py

```diff
@@ -18,22 +18,29 @@
 from openeo.config import get_user_config_dir, get_user_data_dir
 from openeo.util import rfc3339, deep_get, deep_set
 
+if platform.system() == 'Windows':
+    import oschmod
+
+
 _PRIVATE_PERMS = stat.S_IRUSR | stat.S_IWUSR
 
 log = logging.getLogger(__name__)
 
 
 def assert_private_file(path: Path):
     """Check that given file is only readable by user."""
-    mode = path.stat().st_mode
+
+    # use oschmod on Windows
+    # TODO: can we use oschmod for all operating systems, make the code simpler and consitent?
+    if platform.system() == "Windows":
```

Comment:
building on comment above" just do "if oschmod"  here instead of "if windows"

Reply:
and maybe encapsulate the if in a new `get_file_mode` function, which you can also reuse in the tests

Reply:
Done.